misk-aws-dynamodb-testing/src/main/kotlin/misk/aws/dynamodb/testing/DockerDynamoDbModule.kt

```diff
@@ -0,0 +1,68 @@
+package misk.aws.dynamodb.testing
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.client.builder.AwsClientBuilder
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
+import com.amazonaws.services.dynamodbv2.document.DynamoDB
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput
+import com.amazonaws.services.dynamodbv2.model.ResourceInUseException
+import com.google.inject.Provides
+import misk.cloud.aws.FakeAwsEnvironmentModule
+import misk.inject.KAbstractModule
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+
+class DockerDynamoDbModule(
+  private val entities: List<Pair<String, KClass<*>>>
+) : KAbstractModule() {
+  override fun configure() {
+    install(FakeAwsEnvironmentModule())
```

Comment:
This should get installed by the test, not here

Reply:
why is that? It seems odd to me to include a module for something, and it not work because it has another dependancy.

Reply:
That is one exactly what you want with dependency injection. Tests should be isolated and provide its own dependencies. As applied to Guice, modules should only ever install other modules that are part of its own private API.

More practically, if something else installs `FakeAwsEnvironmentModule()`, Guice will throw an exception.